src/Fhir.Anonymizer.Shared.Core/Processors/Settings/PerturbSetting.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using EnsureThat;
+using Fhir.Anonymizer.Core.AnonymizerConfigurations;
+
+namespace Fhir.Anonymizer.Core.Processors.Settings
+{
+    public class PerturbSetting
+    {
+        public double Span { get; set; }
+        public PerturbRangeType RangeType { get; set; }
+        public int RoundTo { get; set; }
+
+        public static PerturbSetting CreateFromRuleSettings(Dictionary<string, object> ruleSettings)
+        {
+            EnsureArg.IsNotNull(ruleSettings);
+
+            var roundTo = 2;
+            if (ruleSettings.ContainsKey(RuleKeys.RoundTo)) 
+            {
+                roundTo = Convert.ToInt32(ruleSettings.GetValueOrDefault(RuleKeys.RoundTo)?.ToString());
+            }
+
+            double span = 0;
+            if (ruleSettings.ContainsKey(RuleKeys.Span))
+            {
+                span = Math.Abs(Convert.ToDouble(ruleSettings.GetValueOrDefault(RuleKeys.Span)?.ToString()));
```

Comment:
Abs here or throw exception?

Reply:
Throw exception is better. Have updated the logic and test cases. Thanks